apps/dashboard/app/models/remote_file.rb

```diff
@@ -0,0 +1,89 @@
+require "rclone_util"
+
+class RemoteFile
+  attr_reader :path, :full_path, :remote
+
+  delegate :basename, :descend, :parent, :join, :to_s, to: :full_path
+
+  def initialize(path, remote)
+    # accepts both String and Pathname
+    # avoids converting to Pathname in every function
+    @path = Pathname.new(path)
+    @remote = remote
+    @full_path = Pathname.new("#{@remote}:#{@path}")
+  end
+
+  def remote_type
+    @remote_type ||= RcloneUtil.remote_type(remote)
+  end
+
+  def raise_if_cant_access_directory_contents
+  end
+
+  def directory?
+    RcloneUtil.directory?(@remote, @path)
+  end
+
+  def ls
+    files = RcloneUtil.ls(@remote, path)
```

Comment:
I think it's a Ruby anti-pattern to use member variables directly outside of an initializer and/or it's own reader/writer.

```suggestion
    files = RcloneUtil.ls(remote, path)
```

Reply:
Good point, fixed.